libs/filer/dbfs_client.go

```diff
@@ -114,7 +228,36 @@ func (w *DbfsClient) Write(ctx context.Context, name string, reader io.Reader, m
 		}
 	}
 
-	handle, err := w.workspaceClient.Dbfs.Open(ctx, absPath, fileMode)
```

Comment:
This new approach really belongs in the SDK.

There is an existing interface for dealing with DBFS files (that is called here).

The details of streaming a file or doing a single put call can be abstracted there and surfaced here with a dedicated `FileMode` to indicate whether it should be a single call or multiple calls.

The size of a file can be retrieved through the `io.Seeker` interface.

The change here should really be limited to determining the file mode and not the implementation.

The SDK guarantees the correctness of the implementation in either streaming or single-call mode.

Reply:
Given that DBFS is a service that will be deprecated, at least the public-facing part I think we should just keep it in the CLI rather than invest time to define the interfaces on the SDK side and using them here.

This PR is really meant to address the regression from the legacy Databricks CLI.